### ISSUE TITLE
Loss of image quality on v2.7.0 vs 2.6.0

### ISSUE THREAD

**Issue by master255** [bk138/droidVNC-NG#240]
@bk138 On the latest version 2.7.2 from the Play market, I got a black bar on the right side of the screen and the image quality deteriorated, even on the maximum image quality setting.
Connection via novnc

This did not happen on version 2.6.0.

The screenshot shows the maximum image quality, but you can clearly see that there are artifacts:
![image](https://github.com/user-attachments/assets/88b66045-3690-4a25-bb07-7bf562d492f1)

**Comment by bk138** [comment]
@master255 There's [no change between 2.7.0 and 2.7.2 affecting image quality](https://github.com/bk138/droidVNC-NG/compare/v2.7.0...v2.7.2) or the black bar issue you're seeing (which is in fact #145, so let's track there :-) ).

**Comment by master255** [comment]
@bk138 You closed the bug too quickly.
My mistake. There were no problems on version 2.6.0. Here's a screenshot. I just installed it. Probably the degradation happened between versions 2.6.0 and 2.7.0
![image](https://github.com/user-attachments/assets/de0f0948-3a03-47f7-9c50-ba03da43d553)

**Comment by master255** [comment]
@bk138 Re-open the bug

**Comment by bk138** [comment]
@master255 here are the changes between 2.6.0 and 2.7.2: https://github.com/bk138/droidVNC-NG/compare/v2.6.0...v2.7.2

Are you able to build from source and do a bisect maybe? If not, having a look at each change would help. 

The only thing I can imagine causing image quality loss is 3da1dda07e1694e54636217987a091b813cbac36 but that was already on 2.4.0.

**Comment by master255** [comment]
@bk138 Too much unfamiliar code. I can test apk files if you provide them and collect logcat logs. I've downloaded the project now and I'm looking at the code. But the project does not compile because of errors in the code.

**Comment by master255** [comment]
@bk138 How can this be fixed? 
![image](https://github.com/user-attachments/assets/d43fcac9-187e-4af7-974f-74643fae736f)

**Comment by bk138** [comment]
@master255 can you compile a list of steps for me to reproduce? With versions used, settings used, outcome, as specific as possible? I can then have a look myself.

**Comment by master255** [comment]
@bk138 Just all defaults and image quality to max.
Android 8.0.
Galaxy S8+.

**Comment by master255** [comment]
@bk138 1. Yes.
2. noVNC buildin
3. 2.6.0 good, 2.7.0+ bad

**Comment by bk138** [comment]
@master255 I now tried 
* v2.6.0 and v2.7.0 
* on the same device,
* built from source,
* with default settings (esp. server-side scaling)
*  and using "fast" screen sharing (i.e. not the fallback mode),
*  connecting via the builtin noVNC:
 
I get the same image quality with both, a black bar on the right is present in both. In short, can't reproduce the issue :-/ 

You might want to check your browser's local storage for noVNC quality settings (if none there, it's using defaults), or try with another browser. Another thing that comes to mind is trying on a different network.

**Comment by master255** [comment]
@bk138 Android 8.0? What's the device?

**Comment by bk138** [comment]
Android 12, some Samsung.

**Comment by master255** [comment]
@bk138 So I'm testing on Android 8.0. You should test on that version too. Or your app is only for Android 12+?

**Comment by master255** [comment]
@bk138 On my Galaxy S22 Ultra Android 12 on version 2.7.2, I can't see the side line. But the image quality is also very poor, even at the highest setting. 
![image](https://github.com/user-attachments/assets/bc8affad-0fa6-4c17-891d-b9cf68700167)

**Comment by master255** [comment]
@bk138 And I'm sure on version 2.6.0 the picture quality will be fine.

**Comment by bk138** [comment]
@master255 can you please test with an Android 8 emulator image and post the setup and results here?

**Comment by master255** [comment]
@bk138 No. I can't test with the emulator because it doesn't have network access

**Comment by bk138** [comment]
> @bk138 No. I can't test with the emulator because it doesn't have network access

--> https://developer.android.com/studio/run/emulator-networking should help

**Comment by master255** [comment]
@bk138 I added port redirection: redir add tcp:5000:6000
Through the emulator console (telnet)
But still can't open the link http://127.0.0.1:5800/vnc.html?autoconnect=true&host=127.0.0.1&port=5900

Maybe I need to do something else?

**Comment by bk138** [comment]
> @bk138 I added port redirection: redir add tcp:5800:5800 redir add tcp:5900:5900
> 
> Through the emulator console (telnet) But still can't open the link http://127.0.0.1:5800/vnc.html?autoconnect=true&host=127.0.0.1&port=5900
> 
> Maybe I need to do something else?

I just tested with 
```
adb forward tcp:5800 tcp:5800
adb forward tcp:5900 tcp:5900
```
Then http://127.0.0.1:5800/vnc.html?autoconnect=true&host=127.0.0.1&port=5900 works.

**Comment by master255** [comment]
@bk138 Yes. I was able to forward the ports too.
I think your problem is that you are overprotecting your program.
Here, I found an obvious problems with the image quality. 
![image](https://github.com/user-attachments/assets/5cf7c7b2-09e1-4580-9ca5-53de6d252329)

**Comment by bk138** [comment]
Well if we can find a fix I'm happy to add it. I'm not saying you don't see an issue. I am just not able to reproduce it nor can I see a recent code change that might trigger it. To get this clear: I hear you, but what can be a proper fix?

**Comment by master255** [comment]
@bk138 It feels like libvncserver has changed. I think the problem is in it.

**Comment by bk138** [comment]
You could try building the server with different commits/versions of libvncserver - within https://github.com/bk138/droidVNC-NG/compare/v2.6.0...v2.7.0 only https://github.com/bk138/droidVNC-NG/commit/d7cf3ac24b2d46e234a2461544142ff113f95092 touches libvncserver, and this is going from https://github.com/LibVNC/libvncserver/commits/acac77968d6ea9a8f09d7b758fe1bffac31da7eb to https://github.com/LibVNC/libvncserver/commits/06cff313d22deebcb55d720541d3faeae2986347 ...

**Comment by master255** [comment]
@bk138 I don't have time to do this.

**Comment by bk138** [comment]
> @bk138 I don't have time to do this. I write other programs. More important ones.

Hey fair enough! Here are two screenshots using the built-in noVNC, server is running on the same Android 8 emulated device:

v2.6.0                      |  v2.7.0
:-------------------------:|:-------------------------:
![grafik](https://github.com/user-attachments/assets/7816813e-31ca-4b25-b8b3-fee169221eb0) | ![grafik](https://github.com/user-attachments/assets/dafc9e4b-1c69-42ee-93e3-f5c31851130b)

tl;dr: can't reproduce. It's probably got something to do with different network conditions, maybe noVNC has got some auto-sensing. Also, as per https://github.com/bk138/droidVNC-NG/issues/240#issuecomment-2362877615, the libvncserver changes in between don't touch any imaging-related code.

**Comment by master255** [comment]
@bk138 Both screens are poor quality. Where is the good quality?
I think you're wrong and I'll send a comparison of the versions on the emulator later.
